tmux-cc/src/lib.rs

```diff
@@ -82,6 +82,12 @@ pub enum Event {
         window: TmuxWindowId,
         name: String,
     },
+    LayoutChange {
+        window: TmuxWindowId,
+        layout_id: String,
+        width: u64,
+        height: u64
+    }
```

Comment:
I'm not sure of their significance/utility, but ideally this struct would have `Option`al fields to hold the `<window_visible_layout>` `<window_raw_flags>` from tmux 3; can you add those too?

Reply:
I don’t think the last two parameters  are really useful at the moment, because the implementation will need to first be capable of handling the old format. I think we can just use the old format now and leave the last two unused for the initial implementation. 

On that note, how do your think wezterm should behave on layout-change, for example the desired layout is smaller than the current window? I want to make tmux-cc happen and I could use some suggestions here. I am very new to the codebase.

Reply:
> I don’t think the last two parameters are really useful at the moment, because the implementation will need to first be capable of handling the old format. I think we can just use the old format now and leave the last two unused for the initial implementation.

Implementation wise, I agree.  The `tmux-cc` crate is just a parser though, so it would be great if we can have it faithfully parse the data out even if we don't use it right away.

> On that note, how do your think wezterm should behave on layout-change, for example the desired layout is smaller than the current window? I want to make tmux-cc happen and I could use some suggestions here. I am very new to the codebase.

If tmux's size is smaller than wezterm's, then I don't think we need to do much that is special: we can pad out the "missing" areas with blanks when we render.

wezterm has a binary tree layout system in the `Tab` struct (in `mux/src/tab.rs`) for keeping track of panes, and one of the challenges of hooking up tmux-cc mode is mapping tmux's panes to it.

The binary tree has some nice properties with respect to constraining the splits for manually adjusting the layout.

In theory, wezterm doesn't really need to know about any of the specifics of tmux's layouts so long as tmux reports the effective size and positions of the panes when the layout changes.  I *think* that all tmux layouts should be able to be mapped to some binary tree (the tmux implementation is itself based on a tree, but not strictly a binary tree), I just haven't had the time to figure out the algorithm for that yet. That mapping operation is logically equivalent to the `Tab::sync_with_pane_tree` method that is used in wezterm's mux protocol, but the difference is that wezterm's mux data is already tree-shaped, whereas the tmux data is a tree rendered as a list.

Reply:
Thanks for all the suggestions! 

I've managed to add support for visiable_layout and flags. I'm new to rust so I might be writing something silly :sweat_smile: